tgui/packages/tgui/interfaces/GeneTek/BioEffect.js

```diff
@@ -68,14 +68,23 @@ export const BioEffect = (props, context) => {
     desc,
     icon,
     research,
-    canResearch,
-    canInject,
-    canScramble,
-    canReclaim,
+    canFlags,
     spliceError,
     dna,
   } = gene;
 
+  const bioEffectCanFlags = {
```

Comment:
Understand the goal/intent to compress here, conscious that it adds complexity for the sake of a couple of extra boolean fields.

I'm fine with changing as you have it as GeneTek needs some harder optimization, but noting for anyone reading that I would prefer maintainability for most simpler UIs.

Reply:
> Consider value of consolidating data at risk of complexity to readability and javscript+dm coupling.

I'm very on board with ripping that out to improve maintainability, if desired.

Reply:
Removed.